test/core/test_fragment.jl

```diff
@@ -1417,3 +1417,116 @@ end
         @test nbonds(res) == nbonds(sys, fragment_idx = res.idx)
     end
 end
+
+@testitem "Fragment/has_torsion" begin
+
+    sys = load_pdb(ball_data_path("../test/data/AlaAla.pdb"))
+    fdb = FragmentDB()
+    build_bonds!(sys, fdb)
+    r1 = residues(sys)[1]
+    r2 = residues(sys)[2]
+
+    @test has_torsion_psi(r1) == true
+    @test has_torsion_psi(r2) == true
+
+    @test has_torsion_phi(r1) == true
+    @test has_torsion_phi(r2) == true
+    @test has_torsion_omega(r1) == true
+    @test has_torsion_omega(r2) == true
+
+    sys2 = deepcopy(sys)
+    @test nfragments(sys2) == 2
+    @test nresidues(sys2) == 2
+    delete!(residues(sys2)[2])
+    @test nresidues(sys2) == 1
+
+    r1 = residues(sys2)[1]
+    @test has_torsion_psi(r1) == false
+    @test has_torsion_phi(r1) == false
+    @test has_torsion_omega(r1) == false
+
+    r1 = residues(sys)[1]
+    r2 = residues(sys)[2]
+
+    @test !has_flag(r1, :N_TERMINAL)
+    @test !has_flag(r1, :C_TERMINAL)
+
+    @test !has_flag(r2, :N_TERMINAL)
+    @test !has_flag(r2, :C_TERMINAL)
+
+    set_flag!(r1, :N_TERMINAL)
+    @test has_flag(r1, :N_TERMINAL)
+
+    set_flag!(r2, :C_TERMINAL)
+    @test has_flag(r2, :C_TERMINAL)
+
+    @test has_torsion_psi(r1) == true
+    @test has_torsion_psi(r2) == false
+
+    @test has_torsion_phi(r1) == false
+    @test has_torsion_phi(r2) == true
+    @test has_torsion_omega(r1) == false
+    @test has_torsion_omega(r2) == true
+
+end
+
+
+@testitem "Fragment/calculate_torsion_angle" begin
+    for T in [Float32, Float64]
+        sys = System{T}()
+        a = Atom(sys, 1, Elements.C, r = Vector3{T}(0.0f0, 1.0f0, 0.0f0))
+        b = Atom(sys, 2, Elements.C, r = Vector3{T}(0.0f0, 1.0f0, -1.0f0))
+        c = Atom(sys, 3, Elements.N, r = Vector3{T}(1.0f0, 0.0f0, 0.0f0))
+        d = Atom(sys, 4, Elements.C, r = Vector3{T}(1.0f0, -1.0f0, -1.0f0))
+
+        @test isapprox(rad2deg(calculate_torsion_angle(a, b, c, d)),149.999, atol=10e-3)
+    end
+end
+
+@testitem "Fragment/calculate_bond_angle" begin
+    sys = load_pdb(ball_data_path("../test/data/AlaAla.pdb"))
+    fdb = FragmentDB()
+    build_bonds!(sys, fdb)
+
+    a = atoms(sys)[1]
+    b = atoms(sys)[2]
+    c = atoms(sys)[3]
+    @test isapprox(rad2deg(calculate_bond_angle(a,b,c)), 109.007, atol=10e-4)
+
+    a.r = b.r
+    @test_logs (:error, "Atoms can't have the same position. No Angle to calculate here.") calculate_bond_angle(a,b,c)
+end
+
+@testitem "Fragment/set_torsion_angle" begin
+
+    for T in [Float32, Float64]
+        sys = System{T}()
+        a = Atom(sys, 1, Elements.C, r = Vector3{T}(0.0, 1.0, 0.0))
+        b = Atom(sys, 2, Elements.C, r = Vector3{T}(0.00, 0.0, 0.0))
+        c = Atom(sys, 3, Elements.N, r = Vector3{T}(1.0, 0.0, 0.0))
+        d = Atom(sys, 4, Elements.C, r = Vector3{T}(1.0, -1.0, 0.0))
+
+        Bond(a, b, BondOrder.Single)
+        Bond(b, c, BondOrder.Single)
+        Bond(c, d, BondOrder.Single)
+
+        @test set_torsion_angle!(a, b, c, d, deg2rad(90.0))
```

Comment:
What's the expected return value of `set_torsion_angle!`? This should also be checked here.